lib/phoenix/config.ex

```diff
@@ -46,18 +46,29 @@ defmodule Phoenix.Config do
   """
   @spec cache(module, term, (module -> {:cache | :nocache, term})) :: term
   def cache(module, key, fun) do
-    case :ets.lookup(module, key) do
-      [{^key, :cache, val}] ->
-        val
-
-      [] ->
-        case fun.(module) do
-          {:cache, val} ->
-            :ets.insert(module, {key, :cache, val})
-            val
-
-          {:nocache, val} ->
-            val
+    try do
+      case :ets.lookup(module, key) do
+        [{^key, :cache, val}] ->
+          val
+
+        [] ->
+          case fun.(module) do
+            {:cache, val} ->
+              :ets.insert(module, {key, :cache, val})
+              val
+
+            {:nocache, val} ->
+              val
+          end
+      end
+    rescue
+      e ->
+        case :ets.info(module) do
+          :undefined ->
+            raise "could not find ets table for endpoint #{inspect(module)}. Make sure your endpoint is started and note you cannot access endpoint functions at compile-time."
+
+          _ ->
+            reraise e, __STACKTRACE__
         end
     end
```

Comment:
We want to keep as little code as possible inside the try. WDYT about this:

```suggestion
    try do
      :ets.lookup(module, key)
    rescue
      e ->
        case :ets.info(module) do
          :undefined ->
            raise "could not find ets table for endpoint #{inspect(module)}. Make sure your endpoint is started and note you cannot access endpoint functions at compile-time."

          _ ->
            reraise e, __STACKTRACE__
        end
    else
      [{^key, :cache, val}] ->
        val

      [] ->
        case fun.(module) do
          {:cache, val} ->
            :ets.insert(module, {key, :cache, val})
            val

          {:nocache, val} ->
            val
        end
    end
```

PS: I have not tested it.